src/components/tl/ucp/tl_ucp_team.c

```diff
@@ -91,11 +91,15 @@ UCC_CLASS_INIT_FUNC(ucc_tl_ucp_team_t, ucc_base_context_t *tl_context,
     }
 
     if (self->topo && !IS_SERVICE_TEAM(self) && self->topo->topo->sock_bound) {
-        max_radix       = ucc_min(UCC_TL_TEAM_SIZE(self),
-                                  ucc_topo_min_socket_size(self->topo));
-
+        max_radix = ucc_min(UCC_TL_TEAM_SIZE(self),
+                            ucc_topo_max_ppn(self->topo) == 1 ?
+                            UCC_TL_TEAM_SIZE(self):
+                            ucc_topo_min_socket_size(self->topo));
+        min_radix = ucc_min(UCC_TL_TEAM_SIZE(self),
+                            ucc_topo_max_ppn(self->topo) == 1 ? 3: 2);
         self->opt_radix = ucc_kn_get_opt_radix(UCC_TL_TEAM_SIZE(self),
-                                               max_radix);
+                                               min_radix, max_radix);
```

Comment:
Semantically speaking, shouldn't the change be inside `ucc_kn_get_opt_radix` only, rather than "tricking" that function by bounding the choice of optimal radix to be greater than 3?
In other words, this patch probably means that the computation of the radix needs to be changed in itself, no?

Reply:
I'm not changing original semantic of this function which is to search optimal radix in specified ranged [min_radix, max_radix]. Ideally all the knomial functions should take team topology into account, not only optimal radix but tree structure itself depends on how ranks are placed.